### PR TITLE
[DDCE-3915] Java 11 + Scala 2.13 upgrade

### PR DESCRIPTION
**Pull request by matthewwardcap**
- Dependencies Updated
- README updated
- Scala 2.13 upgraded to
- Fixed scala warnings

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -20,13 +20,13 @@ object AppDependencies {
     "org.scalatestplus.play"      %% "scalatestplus-play"       % "5.1.0",
     "org.scalatestplus"           %% "scalatestplus-scalacheck" % "3.1.0.0-RC2",
     "wolfendale"                  %% "scalacheck-gen-regexp"    % "0.1.2",
-    "uk.gov.hmrc.mongo"           %% "hmrc-mongo-test-play-28"  % "0.74.0",
```
Comment (gastonracui) [inline project/AppDependencies.scala:23]: same here

**Review by gastonracui — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -3,14 +3,14 @@ import sbt._
 object AppDependencies {
   import play.core.PlayVersion
 
-  val bootstrapVersion = "7.13.0"
+  val bootstrapVersion = "7.15.0"
 
   private val compile: Seq[ModuleID] = Seq(
     play.sbt.PlayImport.ws,
-    "uk.gov.hmrc.mongo" %% "hmrc-mongo-play-28"             % "0.74.0",
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply project/AppDependencies.scala:10]: No issues with `0.74.0` it's just `1.x.x` onwards we need to be careful with

File: project/AppDependencies.scala (hunk 1 above)
Comment (matthewwardcap) [reply project/AppDependencies.scala:23]: No issues with `0.74.0` it's just `1.x.x` onwards we need to be careful with

Comment (matthewwardcap) [reply project/AppDependencies.scala:23]: Rolled back version on recent commit

**Review by gastonracui — APPROVED** [review]

**Pull request merged**